Eatery/Controllers/Eateries/CampusMenuViewController.swift

```diff
@@ -0,0 +1,393 @@
+
+            alertController.addAction(UIAlertAction(title: "Open in Apple Maps", style: .default) { _ in
+                self.openAppleMapsDirections()
+            })
+            
+            alertController.addAction(UIAlertAction(title: "Open in Google Maps", style: .default) { _ in
+                guard let url = URL(string: "comgooglemaps://?saddr=&daddr=\(coordinate.latitude),\(coordinate.longitude)&directionsmode=walking") else {
+                    return
+                }
+                UIApplication.shared.open(url, options: [:], completionHandler: nil)
+            })
+
+            if let presentationController = alertController.popoverPresentationController {
+                presentationController.sourceView = sender
+                presentationController.sourceRect = sender.bounds
+            } else {
+                alertController.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: nil))
+            }
+
+            present(alertController, animated: true, completion: nil)
+        } else {
+            openAppleMapsDirections()
+        }
+    }
+
+    private func openAppleMapsDirections() {
+        let mapItem = MKMapItem(placemark: MKPlacemark(coordinate: eatery.location.coordinate, addressDictionary: nil))
+        mapItem.name = eatery.name
+        mapItem.openInMaps(launchOptions: [MKLaunchOptionsDirectionsModeKey: MKLaunchOptionsDirectionsModeDriving])
+    }
+
+    private func createHeroModifiers(_ groups: HeroModifierGroups...) -> [HeroModifier] {
+        return [.useGlobalCoordinateSpace, .whenPresenting(.delay(0.15))]
+            + (groups.contains(.fade) ? [.fade] : [])
+            + (groups.contains(.translate) ? [.translate(y: 32), .timingFunction(.deceleration)] : [])
+    }
+
+}
+
+extension CampusMenuViewController: PopularTimesViewLayoutDelegate {
+
+    func popularTimesContentSizeDidChange(_ popularTimesView: PopularTimesView) {
+        view.layoutIfNeeded()
+    }
+
+}
+
+extension CampusMenuViewController: TabbedPageViewControllerDelegate {
+
+    func tabbedPageViewController(_ tabbedPageViewController: TabbedPageViewController, titleForViewController viewController: UIViewController) -> String? {
+        guard let mealViewController = viewController as? CampusEateryMealTableViewController else {
+            return nil
+        }
+
+        return mealViewController.meal
+    }
+
+    func tabbedPageViewController(_ tabbedPageViewController: TabbedPageViewController,
+                                  heightOfContentForViewController viewController: UIViewController) -> CGFloat {
+        guard let mealViewController = viewController as? CampusEateryMealTableViewController else {
+            return 0
+        }
+
+        mealViewController.view.layoutIfNeeded()
+        return mealViewController.tableView.contentSize.height
+    }
+    
+}
+
+private class CampusMenuInfoView: UIView {
```

Comment:
i think it might be worth putting this in its own file to prevent this file from getting too long.